cg/meta/workflow/mutant/quality_controller/quality_controller.py

```diff
@@ -0,0 +1,144 @@
+from pathlib import Path
+from cg.meta.workflow.mutant.metrics_parser import MetricsParser, QualityMetrics, SampleMetrics
+from cg.meta.workflow.mutant.quality_controller.models import (
+    SampleQualityResult,
+    CaseQualityResult,
+    QualityResult,
+)
+
+from cg.meta.workflow.mutant.quality_controller.utils import (
+    external_negative_control_pass_qc,
+    get_percent_reads_guaranteed,
+    get_sample_target_reads,
+    is_sample_external_negative_control,
+    is_sample_negative_control,
+    is_valid_total_reads,
+    is_valid_total_reads_for_external_negative_control,
+)
+from cg.store.api.core import Store
+from cg.store.models import Sample
+
+
+class QualityController:
+    def __init__(self, status_db: Store):
+        self.status_db = status_db
+
+    def quality_control(self, case_results_file_path: Path) -> QualityResult:
+        quality_metrics: QualityMetrics = MetricsParser.parse_sample_results(case_results_file_path)
+        sample_results: list[SampleQualityResult] = self.quality_control_samples(quality_metrics)
+        case_result: CaseQualityResult = self.quality_control_case(sample_results)
+        # TODO
+        # report_file: Path = get_report_path(case_metrics_file_path)
+        # ReportGenerator.report(out_file=report_file, samples=sample_results, case=case_result)
+        # ResultLogger.log_results(case=case_result, samples=sample_results, report=report_file)
+        # summary: str = ReportGenerator.get_summary(
+        #     case=case_result, samples=sample_results, report_path=report_file
+        # )
+        # return QualityResult(case=case_result, samples=sample_results, summary=summary)
+
+    def quality_control_samples(self, quality_metrics: QualityMetrics) -> list[SampleQualityResult]:
+        sample_results: list[SampleQualityResult] = []
+        for sample_id, metrics in quality_metrics.samples.items():
+            result = self.quality_control_sample(sample_id=sample_id, metrics=metrics)
+            sample_results.append(result)
+        return sample_results
+
+    def quality_control_sample(self, sample_id: str, metrics: SampleMetrics) -> SampleQualityResult:
+        valid_read_count: bool = self.has_valid_total_reads(sample_id)
+        # valid_10x_coverage: bool = has_valid_10x_coverage(metrics)
+        mutant_qc_pass: bool = self.sample_qc_pass(sample_id, metrics)
+
+        sample: Sample = self.status_db.get_sample_by_internal_id(sample_id)
```

Comment:
Can we just pass the Sample object to the above functions, istead of each of them running self.status_db.get_sample_by_internal_id(sample_id)?